File: stubs/java/src/main/java/com/beastmode/rmddd/examples/ecommerce/OrderLineAddedEvent.java
package com.beastmode.rmddd.examples.ecommerce;

import com.beastmode.rmddd.domain.DomainEvent;

import java.math.BigDecimal;
import java.util.Map;

/**
 * Domain event fired when an order line is added to an order.
 */
public class OrderLineAddedEvent extends DomainEvent {
    
    private final ProductId productId;
    private final int quantity;
    private final BigDecimal unitPrice;
    
    public OrderLineAddedEvent(OrderId orderId, ProductId productId, int quantity, BigDecimal unitPrice) {
        super(orderId);
        this.productId = productId;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
    }
    
    public OrderId getOrderId() {
        return (OrderId) getAggregateId();
    }
    
    public ProductId getProductId() {
        return productId;
    }
    
    public int getQuantity() {
        return quantity;
    }
    
    public BigDecimal getUnitPrice() {
        return unitPrice;
    }
    
    @Override
    public Map<String, Object> getEventData() {
        return Map.of(
            "order_id", getOrderId().getValue().toString(),
            "product_id", productId.getValue().toString(),
            "quantity", quantity,
            "unit_price", unitPrice.toString(),
            "line_total", unitPrice.multiply(BigDecimal.valueOf(quantity)).toString()
        );
    }
}

File: stubs/csharp/src/BeastMode.RmDdd.Examples/Examples/ECommerce/Events/OrderLineAddedEvent.cs
using System.Collections.Generic;
using BeastMode.RmDdd.Domain;

namespace BeastMode.RmDdd.Examples.ECommerce.Events
{
    /// <summary>
    /// Domain event fired when an order line is added to an order.
    /// </summary>
    public class OrderLineAddedEvent : DomainEvent
    {
        public ProductId ProductId { get; }
        public int Quantity { get; }
        public decimal UnitPrice { get; }
        
        public OrderLineAddedEvent(OrderId orderId, ProductId productId, int quantity, decimal unitPrice) 
            : base(orderId)
        {
            ProductId = productId;
            Quantity = quantity;
            UnitPrice = unitPrice;
        }
        
        public OrderId OrderId => (OrderId)AggregateId;
        
        public override IDictionary<string, object> GetEventData()
        {
            return new Dictionary<string, object>
            {
                ["order_id"] = OrderId.Value.ToString(),
                ["product_id"] = ProductId.Value.ToString(),
                ["quantity"] = Quantity,
                ["unit_price"] = UnitPrice,
                ["line_total"] = UnitPrice * Quantity
            };
        }
    }
}